### PR TITLE
fix drop

### PR DESCRIPTION
**Pull request by olafklingt**
in the context of a program i wrote I run into a problem with a segfault on close. because of what happen in drop. 
Here is a draft of a change that solved it for me.

### CODE REVIEW

File: src/client/client_impl.rs [hunk 1]
```diff
@@ -659,6 +659,7 @@ impl Client {
 impl Drop for Client {
     fn drop(&mut self) {
         let _m = CREATE_OR_DESTROY_CLIENT_MUTEX.lock().unwrap();
+        println!("drop client is called");
```
Comment (wmedrano) [inline src/client/client_impl.rs:662]: Please remove this statement.

**Comment by wmedrano** [comment]
lgtm and is generally safer. Can you increment the version number to `0.12.0` as welll?

**Comment by crop2000** [comment]
@wmedrano removed the leftover print statements and updated the version. 
I tried to write a test that triggers the panic i experienced but I didn't succeeded yet.

**Comment by olafklingt** [comment]
Are you interested in a pr that fixes the linting problems?

**Comment by wmedrano** [comment]
If you want to follow up with lint changes go ahead!

**Pull request merged**